File: app/src/main/java/com/disruption/cookcentral/network/RecipeApiService.java
package com.disruption.cookcentral.network;

import com.disruption.cookcentral.models.RecipeResponse;
import com.disruption.cookcentral.models.search.SearchedRecipeResponse;

import io.reactivex.Flowable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RecipeApiService {

    @GET("recipes/random")
    Flowable<RecipeResponse> getRandomRecipes(
            @Query("number") int number,
            @Query("apiKey") String apiKey
    );

    @GET("recipes/search")
    Flowable<SearchedRecipeResponse> searchRecipes(
            @Query("query") String query,
            @Query("number") int number,
            @Query("apiKey") String apiKey
    );
}

File: app/src/main/java/com/disruption/cookcentral/network/RecipeApiService.kt
package com.disruption.cookcentral.network

import com.disruption.cookcentral.models.RecipeResponse
import com.disruption.cookcentral.models.search.SearchedRecipeResponse
import io.reactivex.Flowable
import retrofit2.http.GET
import retrofit2.http.Query

interface RecipeApiService {
    @GET("recipes/random")
    fun getRandomRecipes(
            @Query("number") number: Int,
            @Query("apiKey") apiKey: String?
    ): Flowable<RecipeResponse?>?

    @GET("recipes/search")
    fun searchRecipes(
            @Query("query") query: String?,
            @Query("number") number: Int,
            @Query("apiKey") apiKey: String?
    ): Flowable<SearchedRecipeResponse?>?
}